### PR TITLE
feat(mcp): extract pure utilities and add unit tests

### PR DESCRIPTION
**Pull request by Vswaroop04**
## Summary

* Extracted 8 pure utility functions from the monolithic `packages/mcp/src/index.ts` into a new `src/utils.ts` module:

  * `isJsonResponse`
  * `isLicenseError`
  * `licenseErrorResult`
  * `resolveInstanceId`
  * `buildQuery`
  * `formatProposalText`
  * `getArgValue`
  * `parseErrorResponse`

* Updated `index.ts` to import from `utils.ts` while keeping thin local wrappers around the utilities so all existing call-site signatures remain unchanged and behaviour stays the same.

* Added Vitest to the package along with 40+ unit tests covering:

  * JSON detection edge cases
  * license error parsing
  * instance ID validation
  * query string generation
  * proposal text formatting
  * argument extraction and parsing flows

## Why

The existing `index.ts` file had grown to ~1,260 lines with no automated test coverage. Extracting the stateless logic into a separate module was the smallest safe refactor that made unit testing possible without triggering top-level `await` side effects during imports.

## Test Plan

* [ ] `cd packages/mcp && pnpm install && pnpm test`
* [ ] Verify all tests pass
* [ ] `pnpm build` still succeeds

### CODE REVIEW

**Comment by Vswaroop04** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by KIvanow** [comment]
Hi @Vswaroop04 , thank you for your contribution!

Commit 004f6f56 is unsigned - please rebase and sign before re-review.

Three minor things while you're at it:
 **buildQuery only encodes values, not keys**                                                                                                                                                                                                                                                                                                                                                                                                
  parts.push(`${key}=${encodeURIComponent(String(val))}`);
  Keys are interpolated raw. No practical risk with current callers (all static identifiers), but it's incomplete and the tests don't catch it.                                                                                                                                                                                                                                                                                           
  
  **Misleading test description**
  The 'percent-encodes special characters' test uses FT.SEARCH as its first case - . isn't a reserved URI character and isn't encoded. The actual encoding being tested is a b → a%20b. Worth splitting into two tests or renaming.
                                                                                                                                                                                                                                                                                                                                                                                                                                          
  **resolveInstanceId silently swallows an empty-string override**
  resolveInstanceId(null, '') evaluates '' || null, falls through to null, and throws "No instance selected" - which is a confusing error when an override was explicitly passed. Since the function is now pure and well-tested, this edge case is easy to cover:                                                                                                                                                                        
  if (overrideId !== undefined && overrideId !== null && overrideId !== '') {                                                                                                                                                                                                                                                                                                                                                             
    // use overrideId                                                        
  }                                                                                                                                                                                                                                                                                                                                                                                                                                       
  Or at minimum add a test documenting the current behaviour.

File: packages/mcp/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    (data as Record<string, unknown>).__licenseError === true
+  );
+}
+
+export function licenseErrorResult(data: Pick<LicenseErrorPayload, 'requiredTier' | 'currentTier' | 'upgradeUrl'>): string {
+  return `This feature requires a ${data.requiredTier} license (current tier: ${data.currentTier}). Upgrade at ${data.upgradeUrl}`;
+}
+
+export function resolveInstanceId(activeInstanceId: string | null, overrideId?: string): string {
+  if (overrideId !== undefined && overrideId !== null && overrideId === '') {
```
Comment (KIvanow) [inline packages/mcp/src/utils.ts]: The first 2 checks are redundant because of the third

**Review by KIvanow — CHANGES_REQUESTED** [review]
One small very minor issue with redundant checks

**Comment by Vswaroop04** [comment]
> One small very minor issue with redundant checks

Done

**Comment by KIvanow** [comment]
Nice refactor - the motivation is right, the extraction is type-correct, and the new test coverage is welcome. A few things before this can merge:

1. Regenerate pnpm-lock.yaml. It's the only real merge conflict - index.ts and package.json auto-merge cleanly. pnpm install + commit clears it.
2. Exclude tests from the build. tsconfig.json uses include: ["src/**/*"] with no test exclusion, so pnpm build compiles src/__tests__/utils.test.ts into dist/ and couples the production build to vitest (a devDep) - a --prod build will fail with Cannot find module 'vitest'. Please add "**/*.test.ts" (or "src/__tests__") to exclude, or split out a tsconfig.build.json.
3. The description's "behaviour stays the same" isn't quite accurate. Two extracted functions changed semantics (both intentional and tested - just call them out): buildQuery now URL-encodes keys (was values-only), and resolveInstanceId now throws on an empty-string override (was a silent fallback to the active instance). Both are fine improvements; just update the summary so it's not billed as a pure extraction.
4. Re: my resolveInstanceId guard comment - keeping the three checks is fine; each yields a distinct, more useful error message, so collapsing them into the regex would be a downgrade. Consider it resolved.

Also minor: the test-plan checkboxes are unchecked and the count is 35, not "40+".

Verified locally: 35/35 tests pass, index.ts/utils.ts compile clean, and the merge is clean apart from the lockfile.
